integration/creds/creds_test.go

```diff
@@ -0,0 +1,172 @@
+test "$TEAM_SECRET" = "some team secret"
+
+test "$MIRRORED_VERSION" = "exposed some version not-so-secret"
+
+test "$(cat some-resource/resource_secret)" = "some resource secret"
+test "$(cat custom-resource/custom_resource_secret)" = "some resource secret"
+test "$(cat params-in-get/username)" = "get some username"
+test "$(cat params-in-get/password)" = "get some password"
+test "$(cat params-in-put/version)" = "exposed some version not-so-secret"
+test "$(cat params-in-put/username)" = "put-get some username"
+test "$(cat params-in-put/password)" = "put-get some password"
+
+echo all credentials matched expected values
+`
+
+var pipelineVars = map[string]interface{}{
+	"assertion_script":     assertionScript,
+	"check_failure":        "", // blank so the check doesn't fail
+	"resource_type_secret": "some resource type secret",
+	"resource_secret":      "some resource secret",
+	"job_secret": map[string]interface{}{
+		"username": "some username",
+		"password": "some password",
+	},
+}
+
+var teamVars = map[string]interface{}{
+	"team_secret":      "some team secret",
+	"resource_version": "exposed some version not-so-secret",
+}
+
+func testCredentialManagement(
+	t *testing.T,
+	fly flytest.Cmd,
+	dc dctest.Cmd,
+	setTeamVar func(string, string, interface{}),
+	setPipelineVar func(string, string, string, interface{}),
+) {
+	for name, val := range teamVars {
+		setTeamVar("main", name, val)
+	}
+
+	for name, val := range pipelineVars {
+		setPipelineVar("main", "test", name, val)
+	}
+
+	t.Run("pipelines", func(t *testing.T) {
+		err := fly.Run("set-pipeline", "-n", "-c", "pipelines/credential-management.yml", "-p", "test")
+		require.NoError(t, err)
+
+		err = fly.Run("unpause-pipeline", "-p", "test")
+		require.NoError(t, err)
+
+		t.Run("config is not interpolated", func(t *testing.T) {
+			config, err := fly.Output("get-pipeline", "-p", "test")
+			require.NoError(t, err)
+
+			eachString(pipelineVars, func(val string) {
+				if val != "" {
+					require.NotContains(t, config, val)
+				}
+			})
+
+			eachString(teamVars, func(val string) {
+				if val != "" {
+					require.NotContains(t, config, val)
+				}
+			})
+		})
+
+		t.Run("interpolates resource type checks", func(t *testing.T) {
+			// build will fail if ((check_failure)) doesn't get interpolated to ""
+			err := fly.Run("check-resource-type", "-r", "test/custom-resource-type")
+			require.NoError(t, err)
+		})
+
+		t.Run("interpolates resource checks", func(t *testing.T) {
+			// build will fail if ((check_failure)) doesn't get interpolated to ""
+			err = fly.Run("check-resource", "-r", "test/custom-resource")
+			require.NoError(t, err)
+		})
+
+		t.Run("interpolates job builds", func(t *testing.T) {
+			// build will fail and return err if any values are wrong
+			err := fly.Run("trigger-job", "-w", "-j", "test/some-job")
+			require.NoError(t, err)
+		})
+
+		t.Run("interpolates one-off builds with job inputs", func(t *testing.T) {
```

Comment:
I get

```
missing inputs: some-resource
```

on this test. Not sure why - we trigger `test/some-job` right before, so I'd think it'd have the correct input

Reply:
Weird, lemme take a look. I never saw that failure myself.

Reply:
Shot in the dark: make sure you run `docker-compose build` in the root of the repo. Maybe it was running against an old build?

Reply:
that was after running `docker-compose build`

it's weird though - I disabled the test cleanup, and tried running the same `fly execute` command against the deployment and it worked. will keep investigating

Reply:
We looked into it, and it turns out I'm dumb and need to pull `concourse/dev:latest` every once in a while